### ISSUE TITLE
Latent processes

### ISSUE THREAD

**Issue by SamuelBrand1** [CDCgov/Rt-without-renewal#3]

**Comment by SamuelBrand1** [comment]
_Background concept_: The main logic/workhorse API of `Turing` is the macro `@model`. `@model` is expecting a set of code that looks like a standard function definition but with a stan-like `~` symbol.

The meaning of `X ~ dist` where `dist` is a [distribution](https://juliastats.org/Distributions.jl/stable/) is either:
1. If `X` has been defined with a value, it is treated as an observation with likelihood `pdf(dist, X)`.
2. If `X` has either not been defined or defined as having the value `missing` then it is treated as an unknown parameter/latent variable with prior `dist`.

**NB: this is the same as the numpyro concept of variables having either a determined value (obs data) or value `None` (latent variable)**.

Example:

```julia
@model function my_model(...some args...)
 X ~ some_dist_obj
...some code...
return generated_quantities
end
```

tl; dr version: Turing is a mash up of stan-like syntactic sugar and Numpyro-like data = value/variable = `None` idea.

_Latent process concept_: 

The latent processes that we want to test are each defined as a Turing model. This mean they can be sampled from independently and tested using the standard API to `model` usage.

However, for the overall inference problem against data they are given as arguments to another Turing model which covers the data observation process, and we use the [`@submodel`](https://turinglang.org/DynamicPPL.jl/stable/api/#DynamicPPL.@submodel) macro which allows nesting of Turing models inside other Turing models (honestly this just works).

Reason for this choice is code modularity. We can define and test each model separately, then glue the bits together (and end-to-end test).

**Comment by SamuelBrand1** [comment]
`random_walk` is defined [here](https://github.com/CDCgov/Rt-without-renewal/blob/3c06d369eadd532976a268962b2fb715f5e7f99e/src/latent-processes.jl#L16).

The model defines sampling from a RW std value (default is HalfNormal(0.05)) and generating a RW. Unit test is [here](https://github.com/CDCgov/Rt-without-renewal/blob/main/test/test_latent-processes.jl).

**Comment by SamuelBrand1** [comment]
##### Latent processes to do:

- [ ] RW
- [ ] Spline
- [ ] Difference AR(1)

**Comment by seabbs** [comment]
> Reason for this choice is code modularity. We can define and test each model separately, then glue the bits together (and end-to-end test).

Agree this seems like a natural approach to me and is a good showcase of Julia imo. 

> Latent processes to do:

The current analysis plan (to be discussed) is RW, AR(1), differenced AR(1). Happy to extend that out to splines as well

**Comment by seabbs** [comment]
Closing this in favour of #53 which will address the remaining latent processes we are considering in the first roun.
